supreme/src/commonMain/kotlin/at/asitplus/signum/supreme/crypt/Encryptor.kt

```diff
@@ -0,0 +1,224 @@
+package at.asitplus.signum.supreme.crypt
+
+import at.asitplus.KmmResult
+import at.asitplus.catching
+import at.asitplus.signum.indispensable.AuthTrait
+import at.asitplus.signum.indispensable.Ciphertext
+import at.asitplus.signum.indispensable.SymmetricEncryptionAlgorithm
+import at.asitplus.signum.indispensable.mac.MAC
+import at.asitplus.signum.supreme.mac.mac
+import org.kotlincrypto.SecureRandom
+
+private val secureRandom = SecureRandom()
+
+/**
+ * Creates an encryptor for the specified [secretKey]. Can be used to encrypt arbitrary data.
+ * If no [iv] is specified, a random IV is generated.
+ * [aad] = _Additional Authenticated Data_.
```

Comment:
Maybe write as `@param aad`

Reply:
reformatted to list both params as bullets